assets/js/modules/pagespeed-insights/components/dashboard/DashboardPageSpeed.js

```diff
@@ -51,11 +51,29 @@ const { useSelect, useDispatch } = Data;
 
 export default function DashboardPageSpeed() {
 	const referenceURL = useSelect( ( select ) => select( CORE_SITE ).getCurrentReferenceURL() );
-	const reportMobile = useSelect( ( select ) => select( STORE_NAME ).getReport( referenceURL, STRATEGY_MOBILE ) );
-	const reportDesktop = useSelect( ( select ) => select( STORE_NAME ).getReport( referenceURL, STRATEGY_DESKTOP ) );
 	const strategy = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_DASH_WIDGET, 'strategy' ) ) || STRATEGY_MOBILE;
 	const dataSrc = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_DASH_WIDGET, 'dataSrc' ) ) || DATA_SRC_LAB;
 
+	const {
+		isFetchingMobile,
+		isFetchingDesktop,
+		reportMobile,
+		reportDesktop,
+		errorMobile,
+		errorDesktop,
+	} = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+
+		return {
+			isFetchingMobile: store.isFetchingGetReport( referenceURL, STRATEGY_MOBILE ),
+			reportMobile: store.getReport( referenceURL, STRATEGY_MOBILE ),
+			errorMobile: store.getReportError( referenceURL, STRATEGY_MOBILE ),
+			isFetchingDesktop: store.isFetchingGetReport( referenceURL, STRATEGY_DESKTOP ),
+			reportDesktop: store.getReport( referenceURL, STRATEGY_DESKTOP ),
+			errorDesktop: store.getReportError( referenceURL, STRATEGY_DESKTOP ),
```

Comment:
I don't think we should add another intermediate layer of selectors here. Let's use the raw ones for errors, e.g. `store.getErrorForSelector( 'getReport', [ referenceURL, STRATEGY_DESKTOP ] )`.

Reply:
Updated to use `getErrorForSelector` with appropriate arguments.